libs/firebase-nest/src/lib/firebase-nest.module.ts

```diff
@@ -0,0 +1,60 @@
+import { DynamicModule, Module, OnApplicationBootstrap, OnApplicationShutdown } from '@nestjs/common';
+import { deleteApp, getApps, initializeApp } from '@firebase/app';
+import { initializeApp as initializeAdminApp } from 'firebase-admin/app';
+import { FirebaseConfig } from './interfaces';
+import { FirebaseAuthService } from './services';
+import { FIREBASE_APP, FIREBASE_CONFIG } from './firebase.constants';
+import { FirebaseInterceptor } from './interceptors';
+import * as admin from 'firebase-admin';
+import { FirebaseAuthGuard } from './guards';
+import { HttpModule } from '@nestjs/axios';
+
+const INTERCEPTORS = [FirebaseInterceptor];
+const SERVICES = [FirebaseAuthService];
+const GUARDS = [FirebaseAuthGuard];
+
+@Module({})
+export class FirebaseNestModule implements OnApplicationShutdown, OnApplicationBootstrap {
+  async onApplicationShutdown() {
+    for (const app of getApps()) {
+      const name = app.name;
+      deleteApp(app)
+        .then(function () {
+          console.log('App ' + name + ' deleted successfully');
```

Comment:
Some topic we could dive into in the future: see what the benefits are, if we use Nest's `Logger` instead of `console.log`.